src/components/lib/component/arch/x86_64/cos_asm_simple_stacks.h

```diff
@@ -25,6 +25,57 @@
 
 #define COS_ASM_REQUEST_STACK
 
+#define COS_ULINV_SWITCH_DOMAIN(protdom)			\
+	movl	$protdom,  %ecx;				\
+	addl	%ecx,      %ecx;				\
+	movl	$0b11,     %eax;				\
+	sall	%cl,       %eax;				\
+	notl	%eax;						\
+	andl	$-4,       %eax;				\
+	xor	%rcx,      %rcx;				\
+	xor	%rdx,      %rdx;				\
+	wrpkru;	
+
+/* r13 = tid -> r14 = &stack*/
+#define COS_ULINV_GET_INVSTK					\
+	movabs	$ULK_BASE_ADDR, %r14;				\
+	/* get perthread invstack */				\
```

Comment:
@gparmer @WenyuanShao @evanstella I think this type of line-to-line comment looks better. Plus some general idea above the macro. What do you think?